src/publish/publishers/googlepubsub.rs

```diff
@@ -195,8 +196,9 @@ impl GooglePubSubError {
 pub struct GooglePubSubPublisher<C> {
     identifier: Cow<'static, str>,
     google_cloud_project: Cow<'static, str>,
+    endpoint: Cow<'static, str>,
     client: hyper::Client<C>,
-    authenticator: Arc<Authenticator<C>>,
+    authenticator: Option<Arc<Authenticator<C>>>,
```

Comment:
Please split out the changes related to customizable endpoints to a separate PR.

Reply:
Or at least a separate commit.

Reply:
yeah I'll do that

Reply:
made it a separate commit. I didn't want another PR because that makes branching more complicated for local testing in `standard_hedwig`

Reply:
I'm happy to see this commit merged to the v4.x branch now. (feel free to push directly)

Reply:
Merged that commit, rebased this PR